### ISSUE TITLE
High resolution support

### ISSUE THREAD

**Issue by ManuelGysin** [QuantumBytesInc/RelicsOfAnnorath#67]
# General
High resolution like retina and 4k must be supported.
This issue is the implementation for QuantumBytesInc/RelicsOfAnnorath/issues/53.

## Result

The GUI and all its elements are retina and 4k ready.

## Detailed description 

- Use flexbox 
- Use SVG images

# Dependencies

## Modules

- [ ] ROAG

# Progress

- [ ] Define and create test sequences
- [ ] Implemented task
- [ ] Verify result with test sequences
- [ ] Pull request
- [ ] Merged

**Comment by graphefruit** [comment]
@rotwurstesser1 hows the actual task?

**Comment by rotwurstesser1** [comment]
@graphefruit Alisa is trying to work on it but since  the file we have really is small it isnt easy, pascal also began his try on sunday evening
Alisas version she showed me last time is more detailed and it would be great if we can manage it that way!
Pascal went for a verion where he decided to cut off unnecessary details, but since we dont have a high quality version to start with we might end up going this route either way

**Comment by ManuelGysin** [comment]
Please update.

**Comment by graphefruit** [comment]
This tasks lays down by raphael actualy. @rotwurstesser1  Please provide a current state feedback

**Comment by Sakubik** [comment]
He might still be waiting for me to finish that graphic. I haven't yet.

**Comment by graphefruit** [comment]
@rotwurstesser1 Please provide feedback

**Comment by graphefruit** [comment]
@Sakubik @rotwurstesser1 Please prepare a meeting together and get the graphic together so that @rotwurstesser1 Can test it and we can have a deeper look

**Comment by rotwurstesser1** [comment]
@Sakubik Meeting this Saturday is ok  for you?

**Comment by rotwurstesser1** [comment]
A few cool versions of the input field have been made by @pquantum  
![a1](https://cloud.githubusercontent.com/assets/13884217/26045090/b64039ec-3948-11e7-98e6-ee0c02c3edf5.png)
![a2](https://cloud.githubusercontent.com/assets/13884217/26045089/b63fb008-3948-11e7-866b-de158f6f75a7.png)
![a3](https://cloud.githubusercontent.com/assets/13884217/26045093/b649a022-3948-11e7-8948-ca06f9b01bd8.png)
![a4](https://cloud.githubusercontent.com/assets/13884217/26045091/b640d1ea-3948-11e7-851f-55d5550e8541.png)
![a5](https://cloud.githubusercontent.com/assets/13884217/26045092/b647b488-3948-11e7-8be9-3cc85606d28e.png)
![a6](https://cloud.githubusercontent.com/assets/13884217/26045088/b63e90d8-3948-11e7-9c8e-af864d12cb71.png)

**Comment by rotwurstesser1** [comment]
Workfile
[IllustratorFile_15_5_2017.zip](https://github.com/QuantumBytesInc/RelicsOfAnnorath/files/1000265/IllustratorFile_15_5_2017.zip)

**Comment by rotwurstesser1** [comment]
http://relics.kikoweb.ch/input/

**Comment by rotwurstesser1** [comment]
Next up: 
Define how scaling works, pro and cons

**Comment by graphefruit** [comment]
After meeting with Pascal and Raphael:
First steps looks good so far, now they will migrate the input into a custom directive in angularjs with own `scss`, `js` and `html`

@graphefruit Will provide a custom directive to load .SVG-Graphics on demand which will be included into the DOM, because they will be styled by css-classes. Thats why we can't use background images.

After this custom directive with svg integration is done, the next tasks will be to handle the different states like: 
- Disabled
- Hovered
- Active
- Error

When this is done new elements will come like 
- checkbox
- dropdown
- etc.

**Comment by graphefruit** [comment]
JS-Service and Directive created for creating SVG's on the fly and adding them into the DOM with caching mechanismn.
Needs to be included now from Raphael and Pacal

**Comment by graphefruit** [comment]
@pquantum @rotwurstesser1 You got it working?

**Comment by pquantum** [comment]
@graphefruit Yes it is now implemented in the test environment kikoweb. (http://relics.kikoweb.ch/input/)

**Comment by graphefruit** [comment]
@pquantum Looks good to me!
Like discussed with @rotwurstesser1 - Please add the test font of sebastian aswell. Maybe as label, maybe aswell as inside the input.

Maybe we could provide an attribute to set the styling type?
`input-font="normal"`
or
`input-font="roa"`
e.g.?
this could be accessed by css aswell

**Comment by graphefruit** [comment]
After discussion with @pquantum we're doing some changes.
Elements will get a naming convetion:

- roa-label
- roa-text-input
- roa-svg
- roa-dropdown
- roa....

We'll change the actual behavior of the SVG.
Remove the `svg-src` directive, add a new directive for `roa-svg` and pass there the src attribute and build the SVG.
```
roaSvg
{
  link: {
     attr.svgSrc ...
  }
}
```
@rotwurstesser1  FYI

**Comment by pquantum** [comment]
@graphefruit The following tasks are done:
- All directives are now prefixed with roa. (**roa-text-input** / **roa-svg** / **roa-label**).
- The font from Sebastian is now displayed in an example on the page.
- The roa-svg directive is now a seperate element directive <roa-svg src="..."></roa-svg>instead of an attribute directive.

**Comment by rotwurstesser1** [comment]
Update: Scaling function implemented, rebuilt sass files for better order and easier use, implemented noto basic as test, found better sass compiler and adjusted so we both have same settings
